bundles/org.openhab.binding.juicenet/src/main/resources/OH-INF/thing/juicenet-device.xml

```diff
@@ -0,0 +1,271 @@
+		<item-type>Number:ElectricPotential</item-type>
+		<label>Voltage</label>
+		<description>Current voltage. (V)</description>
+		<state readOnly="true" pattern="%d V"/>
+	</channel-type>
+
+	<channel-type id="whEnergyType">
+		<item-type>Number:Energy</item-type>
+		<label>Current Energy</label>
+		<description>Current power level of vehicle. (Wh)</description>
+		<state readOnly="true" pattern="%.1f kWh"/>
+	</channel-type>
+
+	<channel-type id="savingsType">
+		<item-type>Number</item-type>
+		<label>Savings</label>
+		<description>Current session EV savings.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="wattPowerType">
+		<item-type>Number:Power</item-type>
+		<label>Power</label>
+		<description>Current charging power. (W)</description>
+		<state readOnly="true" pattern="%.2f kW"/>
+	</channel-type>
+
+	<channel-type id="secondsChargingType">
+		<item-type>Number:Time</item-type>
+		<label>Charging Time</label>
+		<description>Charging time since plug-in time. (s)</description>
+		<state readOnly="true" pattern="%.0f min"/>
+	</channel-type>
+
+	<channel-type id="whEnergyAtPluginType">
+		<item-type>Number:Energy</item-type>
+		<label>Energy at Plugin</label>
+		<description>Energy value at the plugging time. (Wh)</description>
+		<state readOnly="true" pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="whEnergyToAddType">
+		<item-type>Number:Energy</item-type>
+		<label>Energy to Add</label>
+		<description>Amount of energy to be added in current session. (Wh)</description>
+		<state pattern="%.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetimeWHEnergyType">
+		<item-type>Number:Energy</item-type>
+		<label>Lifetime Energy</label>
+		<description>Total energy poured to vehicles during lifetime. (Wh)</description>
+		<state readOnly="true" pattern="%,.2f kWh"/>
+	</channel-type>
+
+	<channel-type id="lifetimeSavingsType">
+		<item-type>Number</item-type>
+		<label>Lifetime Savings</label>
+		<description>EV driving saving during lifetime.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="gasCostType">
+		<item-type>Number</item-type>
+		<label>Gas Cost</label>
+		<description>Cost of gasoline used in savings calculations.</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="fuelConsumptionType">
+		<item-type>Number</item-type>
+		<label>Fuel consumption</label>
+		<description>Distance per volume (mpg) used in savings calculations.</description>
+		<state readOnly="true" pattern="%.1f %unit%"/>
+	</channel-type>
+
+	<channel-type id="ecostType">
+		<item-type>Number</item-type>
+		<label>Utility Energy Cost</label>
+		<description>Cost of electricity from utility company. ($/kWh)</description>
+		<state readOnly="true" pattern="$%.2f"/>
+	</channel-type>
+
+	<channel-type id="whPerMileType">
+		<item-type>Number:Power</item-type>
+		<label>Watts per mile</label>
```

Comment:
Seems like it should be:
```suggestion
		<label>Watt Hours Per Mile</label>
```

I'm wondering if this could somehow be made more generic or customizable. It's energy/distance, and for the distance part most European users will probably prefer an SI unit, i.e. km.

Reply:
I looked at this - but I believe we'd have to add Energy / Distance as a unit for QuantityTypes, correct?  Or, do you see another way to do this?

Reply:
I think I need some help here, @openhab/add-ons-maintainers. This would require a new unit like W/m, right?
For reference: https://www.openhab.org/docs/concepts/units-of-measurement.html#list-of-units